File: android/src/main/java/com/bluechilli/flutteruploader/FileItem.java
package com.bluechilli.flutteruploader;

import java.util.Map;

public class FileItem {

  private String fieldname;
  private String path;

  public FileItem(String path) {
    this.path = path;
  }

  public FileItem(String path, String fieldname) {
    this.fieldname = fieldname;
    this.path = path;
  }

  public static FileItem fromJson(Map<String, String> map) {
    return new FileItem(map.get("path"), map.get("fieldname"));
  }

  public String getFieldname() {
    return fieldname;
  }

  public String getPath() {
    return path;
  }
}

File: lib/src/file_item.dart
part of flutter_uploader;

/// Represents a single file in a multipart/form-data upload
class FileItem {
  /// Path to the local file. It is the developers reponsibility to ensure
  /// the path can be accessed.
  final String path;

  /// The field name will be used during HTTP multipart/form-data uploads.
  /// It is ignored for binary file uploads.
  final String field;

  /// Default constructor. The [field] property is set to `file` by default.
  FileItem({
    required this.path,
    this.field = 'file',
  });

  @override
  String toString() => 'FileItem(path: $path fieldname:$field)';

  /// JSON representation for sharing with the underlying platform.
  Map<String, dynamic> toJson() => {'path': path, 'fieldname': field};
}
